### DISCUSSION TITLE
How can I use `CodeChecker parse` in docker `codechecker-web:latest`

### DISCUSSION THREAD

**Discussion by CPythoner** [Ericsson/codechecker#3201] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
When I use `CodeChecker parse` in docker, there is a error `usage: CodeChecker [-h] {cmd,server,store} ...
CodeChecker: error: invalid choice: 'parse' (choose from 'cmd', 'server', 'store')`.

**Describe the solution you would like**
I want to parse my code in docker, How and what should I do ?

**Comment by csordasmarton** [comment, marked as answer]
@CPythoner The [codechecker-web](https://hub.docker.com/r/codechecker/codechecker-web) image is built only from the web part of the CodeChecker. The CodeChecker parse command can be found in the analyze part, so this image doesn't support running this command inside a container.

The main purpose of this image is to run a CodeChecker web server. You can create an build your own image based on our [Dockerfile](https://github.com/Ericsson/codechecker/blob/master/web/docker/Dockerfile). 
In this file you should change this line:
https://github.com/Ericsson/codechecker/blob/4a1fcdfb32edf92303007a86a4bba0908904fb35/web/docker/Dockerfile#L27
to this one to build the whole CodeChecker:
`RUN make -C /codechecker package `

You also have to install other dependencies such as `gcc` to build the logger and you also have to install analyzers (clang, clang-tidy) inside your image.

There is also a [PR](https://github.com/Ericsson/codechecker/pull/2888) which can be useful for you.

Also we have a [snap package](https://snapcraft.io/codechecker) what you can use to install CodeChecker easily. You can install it in your host machine or you can install it inside your docker image.
